lms/djangoapps/ccx/models.py

```diff
@@ -24,6 +26,9 @@ class CustomCourseForEdX(models.Model):
     course_id = CourseKeyField(max_length=255, db_index=True)
     display_name = models.CharField(max_length=255)
     coach = models.ForeignKey(User, db_index=True)
+    # if not empty, this field contains a compressed json serialized list of
+    # the master course modules
+    structure_json = CompressedTextField(verbose_name='Structure JSON', blank=True, null=True)
```

Comment:
Certainly edx-platform has a lot of functionality with no imposed limits, implying that there's no limit to the amount of data that can be added while remaining performant in production at scale - which isn't actually the case. Since this implementation keeps strings in a list and performs set conversions/comparisons/intersections, some threshold of number of modules will cause the code to slow down. My reason for asking for a limit is to block a user from reaching that threshold. I won't insist on it in this case, due to the fact that the number of modules should mirror the course size and the limitless precedent that already exists for course size - you can decide.